Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -818,6 +819,24 @@ internal struct RedirectHandler<ResponseType> {
     }
 
     func redirect(status: HTTPResponseStatus, to redirectURL: URL, promise: EventLoopPromise<ResponseType>) {
+        let nextLimit: HTTPClient.Configuration.RedirectLimit.Next
+        switch self.limit {
+        case .none:
+            nextLimit = .none
+        case .count(let left):
+            if left == 0 {
+                return promise.fail(HTTPClientError.redirectLimitReached)
+            }
+            nextLimit = .count(left: left - 1)
+        case .loop(let visited):
```

Comment:
would it make sense to always fail on loops, rather than make it a configuration option?

Reply:
interesting idea, I made it an option because it requires additional memory to store all visited URL, do you think this is a premature optimization?

Reply:
Loop detection is now default